src/lsptoolshost/restore.ts

```diff
@@ -0,0 +1,108 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { RoslynLanguageServer } from './roslynLanguageServer';
+import { RestorableProjects, RestoreParams, RestorePartialResult, RestoreRequest } from './roslynProtocol';
+import path = require('path');
+
+let _restoreInProgress = false;
+
+export function registerRestoreCommands(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    restoreChannel: vscode.OutputChannel
+) {
+    context.subscriptions.push(
+        vscode.commands.registerCommand('dotnet.restore.project', async (_request): Promise<void> => {
+            return chooseProjectToRestore(languageServer, restoreChannel);
+        })
+    );
+    context.subscriptions.push(
+        vscode.commands.registerCommand('dotnet.restore.all', async (): Promise<void> => {
+            return restore(languageServer, restoreChannel);
+        })
+    );
+}
+async function chooseProjectToRestore(
```

Comment:
The name "choose" made me think this would be returning the project and then calling restore with it or something.